apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -208,13 +212,38 @@ function trackBlocksHandler( blocks, eventName, propertiesHandler = noop, parent
  * @param {string} eventName event name
  * @returns {Function} track handler
  */
-const getBlocksTracker = ( eventName ) => ( blockIds ) => {
+const getBlocksTracker = ( eventName ) => ( blockIds, fromRootClientId, toRootClientId ) => {
 	const blockIdArray = Array.isArray( blockIds ) ? blockIds : [ blockIds ];
 
-	// track separately for each block
-	blockIdArray.forEach( ( blockId ) => {
-		tracksRecordEvent( eventName, { block_name: getTypeForBlockId( blockId ) } );
-	} );
+	const fromContext = getBlockEventContextProperties( fromRootClientId );
+	const toContext = getBlockEventContextProperties( toRootClientId );
+
+	if ( toRootClientId === undefined || isEqual( fromContext, toContext ) ) {
+		// track separately for each block
+		blockIdArray.forEach( ( blockId ) => {
+			tracksRecordEvent( eventName, {
+				block_name: getTypeForBlockId( blockId ),
+				...fromContext,
+			} );
+		} );
+	} else {
+		const fromProps = {};
+		const toProps = {};
+		for ( const key of Object.keys( fromContext ) ) {
+			fromProps[ `from_${ key }` ] = fromContext[ key ];
+		}
+		for ( const key of Object.keys( toContext ) ) {
+			toProps[ `from_${ key }` ] = toContext[ key ];
+		}
```

Comment:
still the same notes as previously listed https://github.com/Automattic/wp-calypso/pull/54352#discussion_r665544515